x-pack/legacy/plugins/security/index.js

```diff
@@ -103,23 +87,22 @@ export const security = (kibana) => new kibana.Plugin({
       }
 
       return {
-        secureCookies: securityPlugin.config.secureCookies,
-        sessionTimeout: securityPlugin.config.sessionTimeout,
+        secureCookies: securityPlugin.__legacyCompat.config.secureCookies,
+        sessionTimeout: securityPlugin.__legacyCompat.config.sessionTimeout,
         enableSpaceAwarePrivileges: server.config().get('xpack.spaces.enabled'),
       };
     },
   },
 
   async postInit(server) {
-    const plugin = this;
-
-    const xpackMainPlugin = server.plugins.xpack_main;
+    const securityPlugin = server.newPlatform.setup.plugins.security;
+    if (!securityPlugin) {
+      throw new Error('New Platform XPack Security plugin is not available.');
+    }
 
-    watchStatusAndLicenseToInitialize(xpackMainPlugin, plugin, async (license) => {
-      if (license.allowRbac) {
-        const { security } = server.plugins;
-        await validateFeaturePrivileges(security.authorization.actions, xpackMainPlugin.getFeatures());
-        await registerPrivilegesWithCluster(server);
+    watchStatusAndLicenseToInitialize(server.plugins.xpack_main, this, async () => {
```

Comment:
__note:__ couldn't keep it in NP for the similar reasons described here https://github.com/elastic/kibana/issues/43456#issuecomment-534520497

Reply:
Right, if https://github.com/elastic/kibana/pull/45796 is accepted then we'll "just" have to chain authorization and authentication code with this and other asynchronous logic we have into giant and unmanageable yarn of promises and observables :rofl: Just kidding, I mean it will take some time until we and core adapt to this change and do everything right.